code/modules/power/supermatter/supermatter_delamination/_sm_delam.dm

```diff
@@ -74,17 +74,46 @@ GLOBAL_LIST_INIT(sm_delam_list, list(
 	return
 
 /// Added to an examine return value.
-/// [/obj/machinery/power/supermatter_crystal/proc/examine]
+/// [/obj/machinery/power/supermatter_crystal/examine]
 /datum/sm_delam/proc/examine(obj/machinery/power/supermatter_crystal/sm)
 	return list()
 
 /// Add whatever overlay to the sm.
-/// [/obj/machinery/power/supermatter_crystal/proc/overlays]
+/// [/obj/machinery/power/supermatter_crystal/update_overlays]
 /datum/sm_delam/proc/overlays(obj/machinery/power/supermatter_crystal/sm)
 	if(sm.final_countdown)
-		return list(mutable_appearance(sm.icon, "causality_field"))
+		return list(mutable_appearance(icon = sm.icon, icon_state = "causality_field", layer = FLOAT_LAYER))
 	return list()
 
+/// Applies filters to the SM.
+/// [/obj/machinery/power/supermatter_crystal/process_atmos]
+/datum/sm_delam/proc/filters(obj/machinery/power/supermatter_crystal/sm)
+	var/new_filter = isnull(sm.get_filter("ray"))
+
+	sm.add_filter(name = "ray", priority = 1, params = list(
```

Comment:
it would be nice to not show this if we have no power. I understand that may not be possible tho

Reply:
We can have it, sort of, with setting the size to 1. 0 looks buggy for some odd reason. LMK if this is unacceptable. 

Same goes for the lights btw, can just modify the clamps if you want it done too.

Reply:
that's fine. if you don't mind that'd be nice yes

Reply:
nvm found a better way

Reply:
> it would be nice to not show this if we have no power. I understand that may not be possible tho

It already did not show if there was no power, it didn't show all the way till around 800.